howso/input_validation.amlg

```diff
@@ -104,4 +104,197 @@
+			invalid_value_found (false)
+		)
+
+		(if initial_features
+			(if single_series
+				;not batch_react_series, initial_values is a 1-d list
+				(map
+					(lambda
+						(if (contains_index !featureDateTimeMap (current_index))
+							(call !CheckDateTimeFormat (assoc
+								datetime_feature (current_index 1)
+								datetime_string (current_value 1)
+								parameter_name "initial_values"
+							))
+						)
+					)
+					(zip initial_features initial_values)
+				)
+
+				;else initial_values is a list of lists
+				(map
+					(lambda
+						(map
+							(lambda
+								(if (contains_index !featureDateTimeMap (current_index))
+									(call !CheckDateTimeFormat (assoc
+										datetime_feature (current_index 1)
+										datetime_string (current_value 1)
+									))
+								)
+							)
+							;zip features to each list of values
+							(zip initial_features (current_value))
+						)
+					)
+					initial_values
+				)
+			)
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in initial_values.") )
+				))
+			)))
+		)
+
+		(if init_time_steps
+			(call !CheckDateTimeFormat (assoc
+				datetime_feature !tsTimeFeature
+				datetime_string (first init_time_steps)
+			))
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in init_time_steps.") )
+				))
+			)))
+		)
+
+		(if final_time_steps
+			(call !CheckDateTimeFormat (assoc
+				datetime_feature !tsTimeFeature
+				datetime_string (first final_time_steps)
+			))
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in final_time_steps.") )
+				))
+			)))
+		)
+
+		(if series_stop_map
+			(map
+				(lambda
+					(if (contains_index !featureDateTimeMap (current_index))
+						(map
+							(lambda
+								(call !CheckDateTimeFormat (assoc
+									datetime_feature (current_index 2)
+									datetime_string (current_value 1)
+								))
+							)
+							;individual stop map: (assoc max __ min __)
+							(current_value)
+						)
+					)
+				)
+				series_stop_map
+			)
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in series_stop_map.") )
+				))
+			)))
+		)
+
+		(if series_stop_maps
+			(map
+				(lambda
+					(map
+						(lambda
+							(if (contains_index !featureDateTimeMap (current_index))
+								(map
+									(lambda
+										(call !CheckDateTimeFormat (assoc
+											datetime_feature (current_index 2)
+											datetime_string (current_value 1)
+										))
+									)
+									;individual stop map: (assoc max __ min __)
+									(current_value)
+								)
+							)
+						)
+						;each series_stop_map
+						(current_value)
+					)
+				)
+				series_stop_maps
```

Comment:
break out of if statement and replace this with:
`(if series_stop_map (list series_stop_map) series_stop_maps)`